File: ofos-client/src/features/wallet/services/walletService.js
import api from '../../../api/axiosConfig';

export const walletService = {
  // Get wallet details
  getWallet: () => {
    return api.get('/wallet');
  },

  // Get wallet balance
  getBalance: () => {
    return api.get('/wallet/balance');
  },

  // Add money to wallet
  addMoney: (amount, paymentMethod) => {
    return api.post('/wallet/add-money', { amount, paymentMethod });
  },

  // Pay with wallet
  payWithWallet: (orderId) => {
    return api.post('/wallet/pay', { orderId });
  },

  // Get transaction history
  getTransactionHistory: (page = 0, size = 10) => {
    return api.get('/wallet/transactions', { params: { page, size } });
  },

  // Get transaction by reference
  getTransactionByReference: (reference) => {
    return api.get(`/wallet/transactions/${reference}`);
  },

  // Request withdrawal
  requestWithdrawal: (amount, upiId) => {
    return api.post('/wallet/withdraw', { amount, upiId });
  },
};

File: online-food-ordering-server/src/main/java/com/ofos/service/WalletService.java
package com.ofos.service;

import com.ofos.dto.request.AddMoneyRequest;
import com.ofos.dto.request.WalletPaymentRequest;
import com.ofos.dto.request.WithdrawRequest;
import com.ofos.dto.response.AddMoneyResponse;
import com.ofos.dto.response.WalletResponse;
import com.ofos.dto.response.WalletTransactionResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.math.BigDecimal;

public interface WalletService {
    
    // Wallet Operations
    WalletResponse getWallet(String userEmail);
    
    BigDecimal getBalance(String userEmail);
    
    AddMoneyResponse addMoney(AddMoneyRequest request, String userEmail);
    
    WalletResponse payWithWallet(WalletPaymentRequest request, String userEmail);
    
    WalletResponse processRefund(Long orderId, BigDecimal amount);
    
    // Transaction History
    Page<WalletTransactionResponse> getTransactionHistory(String userEmail, Pageable pageable);

    Page<WalletTransactionResponse> getAllTransactions(Pageable pageable);
    
    WalletTransactionResponse getTransactionByReference(String reference);
    
    // Withdrawal
    WalletResponse requestWithdrawal(WithdrawRequest request, String userEmail);
    
    // Admin Operations
    void addCashback(Long userId, BigDecimal amount, String reason);
    
    void processFailedTransactions();
}
